bindings/smtp/smtp.go

```diff
@@ -0,0 +1,164 @@
+package smtp
+
+import (
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"strconv"
+
+	"github.com/dapr/components-contrib/bindings"
+	"github.com/dapr/dapr/pkg/logger"
+	"gopkg.in/gomail.v2"
+)
+
+// Mailer allows sending of emails using the Simple Mail Transfer Protocol
+type Mailer struct {
+	metadata Metadata
+	logger   logger.Logger
+}
+
+// Metadata holds standard email properties
+type Metadata struct {
+	Host          string `json:"host"`
+	Port          string `json:"port"`
+	User          string `json:"user"`
+	SkipTLSVerify bool   `json:"skipTLSVerify"`
+	Password      string `json:"password"`
+	EmailFrom     string `json:"emailFrom"`
+	EmailTo       string `json:"emailTo"`
+	EmailCC       string `json:"emailCC"`
+	EmailBCC      string `json:"emailBCC"`
+	Subject       string `json:"subject"`
+}
+
+// NewSMTP returns a new smtp binding instance
+func NewSMTP(logger logger.Logger) *Mailer {
+	return &Mailer{logger: logger}
+}
+
+// Init smtp component (parse metadata)
+func (s *Mailer) Init(metadata bindings.Metadata) error {
+	// parse metadata
+	meta, err := s.parseMetadata(metadata)
+	if err != nil {
+		return err
+	}
+	s.metadata = meta
+
+	return nil
+}
+
+// Operations returns the allowed binding operations
+func (s *Mailer) Operations() []bindings.OperationKind {
+	return []bindings.OperationKind{bindings.CreateOperation}
+}
+
+// Invoke sends an email message
+func (s *Mailer) Invoke(req *bindings.InvokeRequest) (*bindings.InvokeResponse, error) {
+	// Determine host and port
+	port, err := strconv.Atoi(s.metadata.Port)
+	if err != nil {
+		s.logger.Fatal("SMTP binding error: Unable to parse specified port to integer value")
+	}
+	s.logger.Debugf("SMTP binding: using server %v:%v", s.metadata.Host, port)
+
+	// Merge config metadata with request metadata
+	metadata := s.metadata.mergeWithRequestMetadata(req)
+	if metadata.EmailFrom == "" {
+		return nil, fmt.Errorf("SMTP binding error: fromEmail property not supplied in configuration- or request-metadata")
+	}
+	if metadata.EmailTo == "" {
+		return nil, fmt.Errorf("SMTP binding error: emailTo property not supplied in configuration- or request-metadata")
+	}
+	if metadata.Subject == "" {
+		return nil, fmt.Errorf("SMTP binding error: subject property not supplied in configuration- or request-metadata")
+	}
+
+	// Compose message
+	msg := gomail.NewMessage()
+	msg.SetHeader("From", metadata.EmailFrom)
+	msg.SetHeader("To", metadata.EmailTo)
+	msg.SetHeader("CC", metadata.EmailCC)
+	msg.SetHeader("BCC", metadata.EmailBCC)
+	msg.SetHeader("Subject", metadata.Subject)
+	body, _ := strconv.Unquote(string(req.Data))
+	msg.SetBody("text/html", body)
+
+	// Send message
+	dialer := gomail.NewDialer(metadata.Host, port, metadata.User, metadata.Password)
+	if metadata.SkipTLSVerify {
+		/* #nosec */
+		dialer.TLSConfig = &tls.Config{InsecureSkipVerify: true}
+	}
+	if err := dialer.DialAndSend(msg); err != nil {
+		s.logger.Fatal(err)
+	}
+
+	// Log success
+	s.logger.Info("SMTP binding: sent email")
```

Comment:
It was intended as an indication that the email was sent successfully. The To, CC and BCC fields can contain a (potentially large) list of email addresses, so that could amount to a large log entry being written. Also I think it is not right to disclose email addresses in logging. So I would suggest keeping it short and only an indication of success.